### PR TITLE
Bluetooth: Controller: Refactor same peer connection check and add check in Initiator

### PR DESCRIPTION
**Pull request by cvinayak**
Store and check device identity addr type of public or
random.

Add implementation to initiator to check and reject
connection requests to already connected peer.

These changes continue to pass the related 1 peripheral conformance test cases,
and now pass 3 initiator conformance tests.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/ull_filter.c [hunk 1]
```diff
@@ -1103,12 +1103,13 @@ static void conn_rpa_update(uint8_t rl_idx)
 	for (handle = 0U; handle < CONFIG_BT_MAX_CONN; handle++) {
 		struct ll_conn *conn = ll_connected_get(handle);
 
-		/* The RPA of the connection matches the RPA that was just resolved */
-		if (conn &&
-		    conn->peer_addr_type < 2U &&
-		    !memcmp(conn->peer_addr, rl[rl_idx].curr_rpa.val, BDADDR_SIZE)) {
-			memcpy(conn->peer_addr, rl[rl_idx].id_addr.val, BDADDR_SIZE);
-			conn->peer_addr_type += 2U;
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_filter.c:1111]: @wopu-ot What is the reason to store Public or Random ID used for resolvable private address? 1 bit device address type of public or random should suffice for same peer checks, right?

File: subsys/bluetooth/controller/ll_sw/ull_master.c [hunk 2]
```diff
@@ -93,6 +93,18 @@ uint8_t ll_create_connection(uint16_t scan_interval, uint16_t scan_window,
...
+	uint8_t *own_id_addr;
+
+	/* Do not connect twice to the same peer */
+	own_id_addr = ll_addr_get(own_id_addr_type, NULL);
+	if (ull_conn_peer_connected(own_id_addr_type, own_id_addr,
+				    peer_addr_type, peer_addr)) {
+		return BT_HCI_ERR_CONN_ALREADY_EXISTS;
+	}
+#endif /* CONFIG_BT_CTLR_CHECK_SAME_PEER_CONN */
+
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_master.c:107]: @wopu-ot I should resolved the supplied `peer_addr` and store the `peer_id_addr` instead, right?

**Review by cvinayak — COMMENTED** [review]
@wopu-ot Need some clarifications on the privacy scenarios... see inline comments

File: subsys/bluetooth/controller/ll_sw/ull_master.c [hunk 3]
```diff
@@ -93,6 +93,18 @@ uint8_t ll_create_connection(uint16_t scan_interval, uint16_t scan_window,
 		return BT_HCI_ERR_CMD_DISALLOWED;
 	}
 
+#if defined(CONFIG_BT_CTLR_CHECK_SAME_PEER_CONN)
+	uint8_t own_id_addr_type = (own_addr_type & 0x01);
+	uint8_t *own_id_addr;
```
Comment (Thalley) [inline subsys/bluetooth/controller/ll_sw/ull_master.c:98]:
```suggestion
	const uint8_t own_id_addr_type = (own_addr_type & 0x01);
	const uint8_t *own_id_addr;
```

File: subsys/bluetooth/controller/ll_sw/ull_slave.c [hunk 4]
```diff
@@ -112,22 +112,23 @@ void ull_slave_setup(struct node_rx_hdr *rx, struct node_rx_ftr *ftr,
 	link = rx->link;
 
 #if defined(CONFIG_BT_CTLR_CHECK_SAME_PEER_CONN)
-	uint8_t own_addr_type = pdu_adv->rx_addr;
-	uint8_t *own_addr = adv->own_addr;
+	uint8_t peer_id_addr_type = (peer_addr_type & 0x01);
+	uint8_t own_id_addr_type = pdu_adv->rx_addr;
+	uint8_t *own_id_addr = adv->own_id_addr;
```
Comment (Thalley) [inline subsys/bluetooth/controller/ll_sw/ull_slave.c:117]:
```suggestion
	const uint8_t peer_id_addr_type = (peer_addr_type & 0x01);
	const uint8_t own_id_addr_type = pdu_adv->rx_addr;
	const uint8_t *own_id_addr = adv->own_id_addr;
```

**Review by Thalley — APPROVED** [review]
LGTM, but looks like we could add `const` a few more places after the function header was updated

**Review by ppryga-nordic — APPROVED** [review]

**Comment by cvinayak** [comment]
Added DNM to get review from @wopu-ot

**Comment by cvinayak** [comment]
> 
> 
> Added DNM to get review from @wopu-ot

Can yourself @asmk-ot or @mtpr-ot review this one?

**Review by mtpr-ot — APPROVED** [review]
Looks good. Added @erbr-ot to confirm the id address changes.

**Review by erbr-ot — APPROVED** [review]
I tested this solution on our 2.2.1 derivative and it passes our complete bsim test suite which includes the EDTT HCI/GATT/LL tests

**Pull request merged**